### PR TITLE
acquisitions: create module

### PR DESCRIPTION
**Pull request by zannkukai**
* creates a `acquisition` module.
* creates pages to list acquisition accounts using a tree view to
  respect accounts hierarchy.
* creates form to transfer funds between two accounts.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/acquisition/components/account/account-brief-view/account-brief-view.component.html [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, version 3 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (iGormilhit) [inline projects/admin/src/app/acquisition/components/account/account-brief-view/account-brief-view.component.html:16]: The return of the unnecessary weird blank space.

File: projects/admin/src/app/acquisition/components/account/account-detail-view/account-detail-view.component.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, version 3 of the License.
+  
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+  
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (iGormilhit) [inline projects/admin/src/app/acquisition/components/account/account-detail-view/account-detail-view.component.html:16]: Some unnecessary weird blank spaces too.

File: projects/admin/src/app/acquisition/components/account/account-detail-view/account-detail-view.component.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+.amount {
+  font-family: Arial;
```
Comment (iGormilhit) [inline projects/admin/src/app/acquisition/components/account/account-detail-view/account-detail-view.component.scss:19]: Arial, seriously? :smile: 

If it is for numbers, why not just `font-family: 'Courier New', Courier, monospace;`?

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, version 3 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (iGormilhit) [inline projects/admin/src/app/acquisition/components/account/account-list/account-list.component.html:16]: Weird spaces.

File: projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.html [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, version 3 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (iGormilhit) [inline projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.html:16]: Weird spaces.

File: projects/admin/src/app/acquisition/components/order/order-detail-view/order-detail-view.component.html [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, version 3 of the License.
+  
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+  
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (iGormilhit) [inline projects/admin/src/app/acquisition/components/order/order-detail-view/order-detail-view.component.html:15]: Some weird spaces too.

File: projects/admin/src/app/acquisition/components/order/order-detail-view/order-detail-view.component.html [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+-->
+<div *ngIf="record$ | async as record" class="container">
+  <h1 class="mb-3">{{ record.metadata.order_number }}</h1>
+  <section class="my-3 row">
+    <!-- ORDER DETAIL ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ -->
```
Comment (iGormilhit) [inline projects/admin/src/app/acquisition/components/order/order-detail-view/order-detail-view.component.html:20]:
```suggestion
    <!-- ORDER DETAIL (:~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ -->
```
Oh, a jellyfish ! :wink: 
Never mind, it just reminds me https://fr.wikipedia.org/wiki/Telecomix#M%C3%A9duse

**Review by iGormilhit — COMMENTED** [review]
Commit message body: capitalize each first letter of each line.

File: projects/admin/src/app/acquisition/components/account/account-detail-view/account-detail-view.component.scss (hunk 3 above)
Comment (zannkukai) [reply projects/admin/src/app/acquisition/components/account/account-detail-view/account-detail-view.component.scss:19]: I tried, but it's not pretty with current font. 
Arial is also a monospace font, installed on al OS.

File: projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.ts [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  form: FormGroup;
+
+  /** the accounts available for transfer */
+  private _accountsTree: Array<AcqAccount> = [];
+  /** store the selected budgets */
+  private _selectedBudgetPid: string = undefined;
+
+  // GETTER & SETTER ============================================================
+  /** Get the current organisation */
+  get organisation(): any {
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.ts:52]: doc block: missing `@returns`

File: projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.ts [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+    return this._organisationService.organisation;
+  }
+
+  /** Get the currency symbol for the organisation */
+  get currencySymbol(): any {
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.ts:57]: doc block: missing `@returns`

File: projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.ts [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+      this.form.controls.amount.setValidators([
+        Validators.min(0.01),
+        Validators.max(maxTransferAmount)
+      ]);
+    });
+  }
+
+  // PUBLIC FUNCTIONS =========================================================
+  /** get the URL to access account detail view */
+  getDetailUrl(account: AcqAccount): Array<string> {
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/account/account-transfer/account-transfer.component.ts:101]: doc block: missing `@returns`

File: projects/admin/src/app/acquisition/components/order/order-detail-view/order-line/order-line.component.html [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+  </div>
+  <div class="col-3 text-right">
+    <button *ngIf="permissions.update.can" type="button" class="btn btn-outline-primary btn-sm mr-1"
+            [routerLink]="['/records', 'acq_order_lines', 'edit', orderLine.metadata.pid]">
+      <i class="fa fa-pencil"></i>
+    </button>
+
+    <button *ngIf="this.permissions.delete.can; else deleteInfo"
+            type="button" class="btn btn-outline-danger btn-sm"
+            title="{{ 'Delete' | translate}}"
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/order/order-detail-view/order-line/order-line.component.html:50]: missing space before `}`

File: projects/admin/src/app/acquisition/components/order/order-detail-view/order-line/order-line.component.html [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+            (click)="delete.emit(orderLine.metadata.pid)">
+      <i class="fa fa-trash" ></i>
+    </button>
+    <ng-template #deleteInfo>
+      <button type="button" class="btn btn-sm btn-outline-danger disabled"
+              title="{{ 'Delete' | translate}}"
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/order/order-detail-view/order-line/order-line.component.html:56]: missing space before `}`

File: projects/admin/src/app/acquisition/components/order/order-detail-view/order-line/order-line.component.html [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+    <dl class="row">
+      <dt class="col-2 label-title" translate>Account</dt>
+      <dd class="col-10">
+        {{ orderLine.metadata.acq_account.pid | getRecord: 'acq_accounts': 'field': 'name' | async }}
+      </dd>
+      <dt class="col-2 label-title" translate>Status</dt>
+      <dd class="col-10">{{ orderLine.metadata.status | translate }}</dd>
+      <ng-container *ngIf="orderLine.metadata.exchangeRate">
+        <dt class="col-2 label-title" translate>Exchange rate</dt>
+        <dd class="col-10">{{ orderLine.metadata.exchange_rate}}</dd>
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/order/order-detail-view/order-line/order-line.component.html:74]: missing space before `}`

File: projects/admin/src/app/acquisition/pipes/negative-amount.pipe.ts [hunk 14]
```diff
@@ -14,16 +15,15 @@
...
 
-import { TestBed } from '@angular/core/testing';
-import { BudgetTotalService } from './budget-total.service';
+@Pipe({
+  name: 'negativeAmount'
+})
+export class NegativeAmountPipe implements PipeTransform {
 
+  transform(amount: number): number {
+    return 0 - amount;
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/pipes/negative-amount.pipe.ts:26]: `return -Math.abs(num)`

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/acquisition/components/account/account-brief-view/account-brief-view.component.html [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<div class="row account" *ngIf="account && permissions else loading">
+  <!-- NAME ================================================================= -->
```
[the remark this answers is not in the document]
Comment (zannkukai) [reply projects/admin/src/app/acquisition/components/account/account-brief-view/account-brief-view.component.html:19]: These comments are present to visually find easier some blocks. Reduce it decrease the visibility

File: projects/admin/src/app/acquisition/components/account/account-brief-view/account-brief-view.component.ts [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+  @Output() deleteAccount = new EventEmitter();
+
+  /** permission about the record */
+  permissions: any;
+  /** children accounts */
+  children: AcqAccount[] = [];
+
+
+  // GETTER & SETTER ============================================================
+  /** Get the current budget pid for the organisation */
```
[the remark this answers is not in the document]
Comment (zannkukai) [reply projects/admin/src/app/acquisition/components/account/account-brief-view/account-brief-view.component.ts:49]: all you need to know is in the function docstring. If a place a `@return` i will just duplicate the same docstring

**Pull request merged**